pyocd/target/family/target_lpc5500.py

```diff
@@ -158,27 +220,48 @@ def reset_and_halt(self, reset_type=None):
                 base = PERIPHERAL_BASE_S
             else:
                 base = PERIPHERAL_BASE_NS
-            
-            # Use the flash programming model to check if the first flash page is readable, since
-            # attempted accesses to erased pages result in bus faults. The start and stop address
-            # are both set to 0x0 to probe the sector containing the reset vector.
+
+            #
+            # Check to see if the flash is erased
+            #
             self.write32(base + FLASH_STARTA, 0x00000000) # Program flash word start address to 0x0
             self.write32(base + FLASH_STOPA, 0x00000000) # Program flash word stop address to 0x0
-            self.write_memory_block32(base + FLASH_DATAW0, [0x00000000] * 8) # Prepare for read
             self.write32(base + FLASH_INT_CLR_STATUS, 0x0000000F) # Clear Flash controller status
-            self.write32(base + FLASH_CMD, FLASH_CMD_READ_SINGLE_WORD) # Read single flash word
+            self.write32(base + FLASH_CMD, FLASH_CMD_BLANK_CHECK) # Check if page is cleared
 
             # Wait for flash word read to finish.
             with timeout.Timeout(5.0) as t_o:
                 while t_o.check():
                     if (self.read32(base + FLASH_INT_STATUS) & 0x00000004) != 0:
                         break
                     sleep(0.01)
-            
+
             # Check for error reading flash word.
             if (self.read32(base + FLASH_INT_STATUS) & 0xB) == 0:
-                 # Read the reset vector address.
-                reset_vector = self.read32(0x00000004)
+                LOG.info("required flash area is erased")
+                halt_only = True
+
+            # Use the flash programming model to check if the first flash page is readable, since
+            # attempted accesses to erased pages result in bus faults. The start and stop address
+            # are both set to 0x0 to probe the sector containing the reset vector.
+            self.write32(base + FLASH_STARTA, 0x00000000) # Program flash word start address to 0x0
+            self.write32(base + FLASH_STOPA, 0x00000000) # Program flash word stop address to 0x0
```

Comment:
Does using sector 0 also work if the device boots in Secure mode, with the initial VTOR_S set to 0x1000_0000?

Reply:
I'm not using secure mode so I'm not sure what should happen. The original code is reading the vector from hardcoded 0 address, that's what I'm using here. I think secure mode support would be out of scope for this issue.

Reply:
Secure mode being out of scope for this PR is totally reasonable.